nmdc_notebook_tools/functional_search.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+from nmdc_notebook_tools.collection_search import CollectionSearch
+
+
+class FunctionalSearch:
+    """
+    Class to interact with the NMDC API to filter functional annotations by KEGG, COG, or PFAM ids.
+    """
+
+    def __init__(self):
+        self.collectioninstance = CollectionSearch("functional_annotation_agg")
+
+    def get_functional_annotation_id(
+        self, id: str, id_type: str, page_size=25, fields="", all_pages=False
+    ):
+        """
+        Get a record from the NMDC API by id. ID types can be KEGG, COG, or PFAM.
+        params:
+            id: str
+                The data base id to query the function annotations.
```

Comment:
Can we re-name the id parameter to "annotation" and id_type to "annotation_type".  These aren't ids in the sense of all the other classes, so I'd like to not use that language.